Add in a service name detector to the ResourceConfigurator.cs (#2219)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/iis-instrumentation.md

For ASP.NET application you can configure the most common `OTEL_` settings
(like `OTEL_SERVICE_NAME`) via `appSettings` in `Web.config`.

> If a service name is not provided manually by environment variables or in the config, one will be generated for you.

Reviewer: ```suggestion
If a service name is not explicitly configured, one will be generated for you.
```
